chore: split out Java connector IAM Auth sample and region tags

shubha-rajan opened the pull request:
- [ ] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] API's need to be enabled to test (tell us)
- [ ] Environment Variables need to be set (ask us to set them)
- [ ] **Tests** pass:   `mvn clean verify` **required**
- [ ] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] Please **merge** this PR for me once it is approved.

enocom commented on cloud-sql/postgres/servlet/src/main/java/com/example/cloudsql/ConnectorIamAuthConnectionPoolFactory.java:
```diff
@@ -0,0 +1,75 @@
+  // keep secrets safe.
+  private static final String INSTANCE_CONNECTION_NAME =
+      System.getenv("INSTANCE_CONNECTION_NAME");
+  private static final String INSTANCE_UNIX_SOCKET = System.getenv("INSTANCE_UNIX_SOCKET");
```
This is unused

shubha-rajan replied:
removed!

enocom reviewed (commented):
We probably should check for a `DB_IAM_USER` in the main application code and use the IAM factory if it's set.

Also, nit: should it be `ConnectorIamAuthNConnectionPoolFactory`?

shubha-rajan commented:
> We probably should check for a `DB_IAM_USER` in the main application code and use the IAM factory if it's set.
> 
> Also, nit: should it be `ConnectorIamAuthNConnectionPoolFactory`?

Renamed the class and added a check for DB_IAM_USER in ConnectionPoolContextListener.
